src/js_results.hpp

```diff
@@ -96,20 +95,13 @@ typename T::Object ResultsClass<T>::create_instance(ContextType ctx, SharedRealm
 }
 
 template<typename T>
-typename T::Object ResultsClass<T>::create_instance(ContextType ctx, SharedRealm realm, const ObjectSchema &object_schema, Query query, bool live) {
-    auto results = new realm::Results(realm, object_schema, std::move(query));
-    results->set_live(live);
-
-    return create_object<T, ResultsClass<T>>(ctx, results);
-}
-
-template<typename T>
 template<typename U>
 typename T::Object ResultsClass<T>::create_filtered(ContextType ctx, const U &collection, size_t argc, const ValueType arguments[]) {
-    auto query_string = Value::validated_to_string(ctx, arguments[0], "predicate");
-    auto query = collection.get_query();
     auto const &realm = collection.get_realm();
     auto const &object_schema = collection.get_object_schema();
+    auto query = collection.get_table()->where();
```

Comment:
This is making the assumption that a collection always has an associated table. That's not true of a default-constructed `Results` or `List` instance. This may not be a problem in practice as realm-js appears to make the same assumption about collections always having an object schema, but I figured I'd mention it.